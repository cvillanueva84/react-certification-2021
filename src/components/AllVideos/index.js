export { default } from './AllVideos.component';