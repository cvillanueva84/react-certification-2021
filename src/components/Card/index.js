export {default} from './Card.component.jsx'; 