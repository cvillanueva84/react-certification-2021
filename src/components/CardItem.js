import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

