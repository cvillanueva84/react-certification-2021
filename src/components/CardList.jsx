import React from 'react';
import Grid from '@material-ui/core/Grid';
import VideoCard from './VideoCard';
import videos from '../youtube-videos-mock';

export default function CardList() {
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      style={{ padding: '30px' }}
    >
      {videos.items.map((video) => (
        <VideoCard videoInfo={video.snippet} key={video.etag} />
      ))}
    </Grid>
  );
}
