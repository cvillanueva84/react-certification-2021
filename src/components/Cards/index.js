export { default } from './Cards.component';
