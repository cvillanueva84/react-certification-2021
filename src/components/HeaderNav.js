import React, { useState, useContext } from 'react';
import { GlobalContext } from './Context';
import { useHistory } from 'react-router-dom';
import { HeaderNavUi } from './ui/HeaderNavUi';



export const HeaderNav = () => {
  const { myStateReducer, dispatch } = useContext(GlobalContext);
  const [inputValue, setValue] = useState('');
  const history = useHistory();

  const handleInputText = (e) => {
    setValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch({
      type: 'actionAddSearch',
      payload: {
        ...myStateReducer,
        search: inputValue,
      },
    });
    setValue('');
  };

  const lightMode = () => {
    dispatch({
      type: 'actionLightMode',
      payload: {
        ...myStateReducer,
      },
    });
  };

  const darkMode = () => {
    dispatch({
      type: 'actionDarkMode',
      payload: {
        ...myStateReducer,
      },
    });
  };

  const handleLogout = () => {
    dispatch({
      type: 'actionLogout',
      payload: {
        ...myStateReducer,
        mode: 'light',
      },
    });

    history.replace('/login');
  };

  return (
    <HeaderNavUi
      handleInputText={handleInputText}
      handleSubmit={handleSubmit}
      lightMode={lightMode}
      darkMode={darkMode}
      handleLogout={handleLogout}
      myStateReducer={myStateReducer}
      inputValue={inputValue}
    />
  );
};
