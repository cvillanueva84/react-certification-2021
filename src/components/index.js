export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
