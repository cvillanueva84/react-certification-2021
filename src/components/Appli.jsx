import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
