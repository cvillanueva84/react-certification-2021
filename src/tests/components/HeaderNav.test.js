import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router';
import { HeaderNav } from '../../components/HeaderNav';
import { GlobalContext } from '../../components/Context';
//import { render, screen, fireEvent } from "@testing-library/react";

describe('Test HeaderNav', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {}
        }
    }
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <HeaderNav/>
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });
    
    it('should render success', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });

    it('should have 3 bottons', () => {
        expect(wrapper.find('button').length).toBe(4);
    });

    it('should have left-header', () => {
        expect(wrapper.getElements("left-header").length).toBe(1);
    });

    it('should have right-header', () => {
        expect(wrapper.getElements("left-right").length).toBe(1);
    });

    it('should have box_input', () => {
        expect(wrapper.getElements("box_input").length).toBe(1);
    });

    it('should mode light', () => {
        wrapper.find('.toggle_left').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalled();
    });

    it('should mode dark', () => {
        wrapper.find('.toggle_right').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalled();
    });

    it('should Submit', () => {
        wrapper.find('input').simulate('submit');
        expect(contextValue.dispatch).toHaveBeenCalled();
    });

    it('should Logout', () => {
        wrapper.find('#logout').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalled();
    });


    /*
    it('should has 4 buttons', () => {
        render(<HeaderNav />);
        expect(screen.getAllByRole("button").length).toBe(4);
    });

    it('should has placeholder Search...', () => {
        render(<HeaderNav />);
        expect(screen.getAllByPlaceholderText("Search...").length).toBe(1);
    });

    it('should has sun', () => {
        render(<HeaderNav />);
        expect(screen.getAllByText("ðŸŒž").length).toBe(1);
    });

    it('should has moon', () => {
        render(<HeaderNav />);
        expect(screen.getAllByText("ðŸŒœ").length).toBe(1);
    });

    
    it('should click', () => {
        const handleClick = jest.fn();
        render(<button onClick={handleClick}></button>);
        fireEvent.click(screen.getAllByText("ðŸŒž"));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });*/

});