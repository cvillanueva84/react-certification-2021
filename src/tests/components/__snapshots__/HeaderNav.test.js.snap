// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HeaderNav should render success 1`] = `ReactWrapper {}`;
